src/seedu/addressbook/commands/FindCommand.java

```diff
@@ -50,11 +50,29 @@ public CommandResult execute() {
         final List<ReadOnlyPerson> matchedPersons = new ArrayList<>();
         for (ReadOnlyPerson person : addressBook.getAllPersons()) {
             final Set<String> wordsInName = new HashSet<>(person.getName().getWordsInName());
-            if (!Collections.disjoint(wordsInName, keywords)) {
+            if (hasSameCharacters(wordsInName, keywords)) {
                 matchedPersons.add(person);
             }
         }
         return matchedPersons;
     }
 
+    /**
+     * Converts the Strings in the String Sets to lower case for comparison, which will allow the search to be case-insensitve.
+     *
+     * @param String set and set2 for searching
+     * @return boolean value of whether there is a subject that has the same name as the input keywords, set
+     */
+    private static boolean hasSameCharacters(Set<String> set, Set<String> set2) {
```

Comment:
Instead of naming it `set` and `set2`, may be `words` and `wordsInKeywords` are better.